alphalens/utils.py

```diff
@@ -205,29 +211,40 @@ def compute_forward_returns(factor_idx,
     Returns
     -------
     forward_returns : pd.DataFrame - MultiIndex
-        Forward returns in indexed by date and asset.
-        Separate column for each forward return window.
+        A MultiIndex DataFrame indexed by timestamp (level 0) and asset
+        (level 1), containing the forward returns for assets.
+        Forward returns column names must follow the format accepted by
+        pd.Timedelta (e.g. '1D', '30m', '3h15m', '1D1h', etc).
+        'date' index freq property must be set to a trading calendar (pandas
+        DateOffset), see infer_trading_calendar for more details.
+        This information is currently used only in cumulative returns
+        computation
     """
 
-    factor_idx = factor_idx.intersection(prices.index)
+    factor_dateindex = factor.index.levels[0]
+    if factor_dateindex.tz != prices.index.tz:
+        raise NonMatchingTimezoneError("The timezone of 'factor' is not the "
+                                       "same as the timezone of 'prices'. See "
+                                       "the pandas methods tz_localize and "
+                                       "tz_convert.")
 
-    if len(factor_idx) == 0:
-        raise ValueError("Factor and prices indices don't match: make sure "
-                         "they have the same convention in terms of datetimes "
-                         "and symbol-names")
+    freq = infer_trading_calendar(factor_dateindex, prices.index)
 
+    factor_dateindex = factor_dateindex.intersection(prices.index)
```

Comment:
Here we need to add a check that got lost during the rebasing:

    if len(factor_dateindex) == 0:
        raise ValueError("Factor and prices indices don't match: make sure "
                         "they have the same convention in terms of datetimes "
                         "and symbol-names")